### PR TITLE
Split misc routines from math.cc

### PR DESCRIPTION
**Pull request by asi1024**
Part of #7143.
~Depends on #7261.~
~Depends on #7270.~

### CODE REVIEW

File: chainerx_cc/chainerx/native/native_device/misc.cc [hunk 1]
```diff
@@ -7,51 +7,27 @@
...
-        VisitFloatingPointDtype(out.dtype(), [&](auto pt) {
-            using T = typename decltype(pt)::type;
-            struct Impl {
-                void operator()(int64_t /*i*/, T x, T& out) { out = chainerx::Sqrt(x); }
-            };
-            Elementwise<const T, T>(Impl{}, x_cast, out);
-        });
-    }
-};
+CHAINERX_NATIVE_REGISTER_ELTWISE_FLOAT_UNARY_KERNEL(CeilKernel, { out = chainerx::Ceil(x); });
```
Comment (niboshi) [inline chainerx_cc/chainerx/native/native_device/misc.cc]: I think `ceil` and `floor` should be in `rounding`.

Comment (asi1024) [reply chainerx_cc/chainerx/native/native_device/misc.cc]: I'll work on it in another PR.

Comment (niboshi) [reply chainerx_cc/chainerx/native/native_device/misc.cc]: Then why did you move this in this PR?

Comment (asi1024) [reply chainerx_cc/chainerx/native/native_device/misc.cc]: These kernels were defined here before this PR. This PR just rewrites the definitions of kernels in `*_device/misc.cc` using macro. Should we revert them?

Comment (niboshi) [reply chainerx_cc/chainerx/native/native_device/misc.cc]: OK, I understand, but it was difficult to notice in the review. Maybe it's better to separate the fix, or at least leave some comment.

File: chainerx_cc/chainerx/routines/misc.cc [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+#include "chainerx/backward_builder.h"
+#include "chainerx/backward_context.h"
+#include "chainerx/dtype.h"
+#include "chainerx/error.h"
+#include "chainerx/graph.h"
+#include "chainerx/kernels/misc.h"
+#include "chainerx/macro.h"
+#include "chainerx/routines/creation.h"
+#include "chainerx/routines/logic.h"
+// #include "chainerx/routines/manipulation.h"
```
Comment (niboshi) [inline chainerx_cc/chainerx/routines/misc.cc]: Remove it

File: chainerx_cc/chainerx/routines/misc.cc [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+#include "chainerx/routines/routines_util.h"
+// #include "chainerx/routines/statistics.h"
```
Comment (niboshi) [inline chainerx_cc/chainerx/routines/misc.cc]: Ditto

**Comment by niboshi** [comment]
Jenkins, test this please

File: chainerx_cc/chainerx/routines/misc.cc (hunk 2 above)
Comment (asi1024) [reply chainerx_cc/chainerx/routines/misc.cc]: Fixed.

**Comment by niboshi** [comment]
Jenkins, test this please

**Pull request merged**